### DISCUSSION TITLE
[Question] How to install Crosswire Sword modules offline?

### DISCUSSION THREAD

**Discussion by AndersonTorres** [ezra-bible-app/ezra-bible-app#727] [Q&A] [answered]
Hello, people! I believe it is a newbie question, but I will do anyway.

Is it possible to download the modules directly from [Crosswire page](https://www.crosswire.org/sword/modules/index.jsp) and use them offline in Ezra app, say in another computer without Internet connection?

**Comment by tobias-klein** [comment, marked as answer]
Dear @AndersonTorres,

Yes, this is possible. Ezra Bible App uses the standard directories for SWORD on Windows/Linux/macOS.

Some information is available here:
http://crosswire.org/sword/modules/moduleinstall.jsp

Aside from that - here is how you can find out the storage location of the SWORD modules on your computer:

1. Open the info dialog in Ezra Bible App (click on the i symbol).
2. Go to the tab "Application Info"
3. Scroll down to "Versions and paths"
4. The path is shown in the line "SWORD module path".

I hope that helps!

**Comment by lxs602** [comment]
Is it possible for Android (13.0)? I don't seem to be able to navigate to /data/user/0/net.ezrabibleapp.cordova

**Reply by tobias-klein** [reply]
Hi @lxs602!

This is unfortunately not supported starting from Android 11. Arbitrary sdcard directory access has become very difficult in the most recent Android versions and as a developer I pretty much gave up on it. Sorry for that.

Something I had considered, though is to implement a feature that syncs the SWORD module folder with a cloud folder (like Dropbox). Could that be an alternative for you?

**Comment by lxs602** [comment]
Sounds like a good workaround. I have heard permissions are difficult on Android. Would this be any better for users for whom persecution through tracking is a concern? Also, would access to an additional sdcard (rather than internal memory), or to a USB, be the same?  Are you still in need of beta testers for Android?

**Comment by lxs602** [comment]
How about installing from a remote URL?
